com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderProgramBuilder.java

```diff
@@ -81,9 +94,12 @@ static public String compileGLSL(String shaderSource, ES2ToES3Converter.ShaderTy
         String firstNonDirectiveLine = null;
         int directiveLineCount = 0;
 
-        Pattern directiveLinePattern = Pattern.compile("^\\s*(#|//).*");
+        Pattern directiveLinePattern = Pattern.compile("^\\s*(#).*");
         Scanner scanner = new Scanner(shaderSource);
 
+        // Iterate the source lines to find the first occurance of a 'valid' line
+        // i.e a line that doesn't already have a preprocessor directive, or is empty
+        // This is needed to get a valid line number when shader compilation has failed
```

Comment:
This (and the lines after) is used for getting debug output if the shader fails to compile in runtime and won't work 100% with pasting the include source since most includes will use #ifdef header guards and the lines won't correspond to the actual lines from the includes.

But on the other hand.. it already produces wrong #line pragmas right now anyway if you have block comments for instance, so it's already incorrect behaviour anyway and I don't know what the alternative is :(

One really roundabout way of doing it would be to split the source by each include since OpenGL supports uploading shader source by a list of strings (https://registry.khronos.org/OpenGL-Refpages/gl4/html/glShaderSource.xhtml) and then using this information to output which include the error came from. I don't think it should be a showstopper for this feature, but perhaps something that will come up eventually..